### PR TITLE
Position lastMsgButton with constraints

### LINKED ISSUE #921: [Bug]: Auto-scroll button in wrong position when switching chats
### What Monal Release channel are you using?

Beta

### iOS system version

16.2

### iOS Monal version

6.0.0

### macOS system version

_No response_

### macOS Monal version

_No response_

### Used XMPP server (domain)

yax.im

### Which XMPP-Server software are you using?

None

### XMPP Server Software Version

_No response_

### How many accounts are you using in Monal?

1

### What happened?

When switching between chats, the auto-scroll button changes position making the user unable to click it. 

### Anything else?

_No response_

### FAQ

- [X] I have [checked the FAQ](https://github.com/monal-im/Monal/wiki/FAQ---Frequently-Asked-Questions) for my issue

### Considerations for XMPP users

- [X] I have checked if my issue can be solved with [Considerations for XMPP users](https://github.com/monal-im/Monal/wiki/Considerations-for-XMPP-users) and [Considerations for XMPP server admins](https://github.com/monal-im/Monal/wiki/Considerations-for-XMPP-server-admins)

### Related Issues

- [X] I have cross-checked this overview https://github.com/monal-im/Monal/issues/322 as well as filtered for related labels https://github.com/monal-im/Monal/labels

### XEP-Check

- [X] I have checked that at least XEP-198, XEP-0280, XEP-0352, XEP-0357, XEP-0313 and XEP-0163 are activated on my server and shown as 'green' under Settings --> Account--> (i) in advanced settings

### Notifications-Menu

- [X] I have checked that all checkmarks are present under Settings --> Notifications

### PR DESCRIPTION
**Pull request by matthewrfennell**
https://github.com/monal-im/Monal/assets/148400980/85f43684-9a4b-4c0c-912a-2203ff3bc4d8

fixes #921

### CODE REVIEW

**Comment by tmolitor-stud-tu** [comment]
That button seems to be glued to the right edge of the screen. Could you please move it a bit to have equal distance from the bottom input field and the right screen edge?

Other than that your PR looks quite good! :)

File: Monal/Classes/chatViewController.m [hunk 1]
```diff
@@ -347,12 +348,29 @@ -(void) initAudioRecordButton
...
+    [self.inputContainerView addConstraints:@[
+        [NSLayoutConstraint constraintWithItem:self.lastMsgButton
+                                     attribute:NSLayoutAttributeTrailing
+                                     relatedBy:NSLayoutRelationEqual
+                                        toItem:self.inputContainerView
+                                     attribute:NSLayoutAttributeTrailing
+                                    multiplier:1.0
+                                      constant:-LAST_MSG_BUTTON_OFFSET],
+        [NSLayoutConstraint constraintWithItem:self.lastMsgButton
+                                     attribute:NSLayoutAttributeBottom
```
Comment (tmolitor-stud-tu) [inline Monal/Classes/chatViewController.m:363]: The colons aren't aligned. Please either align the colons or use only one indentation depth for all arguments having their own row.

**Review by tmolitor-stud-tu — CHANGES_REQUESTED** [review]

File: Monal/Classes/chatViewController.m (hunk 1 above)
Comment (matthewrfennell) [reply Monal/Classes/chatViewController.m:363]: > or use only one indentation depth for all arguments having their own row

Just to make sure I've understood, would this be ok?

```objective-c
[self.inputContainerView addConstraints:@[
    [NSLayoutConstraint
        constraintWithItem:self.lastMsgButton
        attribute:NSLayoutAttributeTrailing
        relatedBy:NSLayoutRelationEqual
        toItem:self.inputContainerView
        attribute:NSLayoutAttributeTrailing
        multiplier:1.0
        constant:-LAST_MSG_BUTTON_OFFSET
    ],
    [NSLayoutConstraint
        constraintWithItem:self.lastMsgButton
        attribute:NSLayoutAttributeBottom
        relatedBy:NSLayoutRelationEqual
        toItem:self.inputContainerView
        attribute:NSLayoutAttributeTop
        multiplier:1.0
        constant:-LAST_MSG_BUTTON_OFFSET
   ],
]];
```

**Comment by matthewrfennell** [comment]
> That button seems to be glued to the right edge of the screen. Could you please move it a bit to have equal distance from the bottom input field and the right screen edge?

I've only tested on iPhone 15 Pro and iPad 10th Generation simulators - on these the position looks unchanged from develop to me. Am I missing something?

Either way though, happy to adjust. I tried some options, which do you prefer?

Develop:

develop

Current PR:

current-pr

Current PR - 5

current-pr-minus-5

Current PR - 10

current-pr-minus-10

> Other than that your PR looks quite good! :)

Thanks, and thanks a lot for such a quick review :)

File: Monal/Classes/chatViewController.m (hunk 1 above)
Comment (tmolitor-stud-tu) [reply Monal/Classes/chatViewController.m:363]: yes, either this, or that (on your discretion):

```
[NSLayoutConstraint
        constraintWithItem:self.lastMsgButton
                         attribute:NSLayoutAttributeTrailing
                       relatedBy:NSLayoutRelationEqual
                            toItem:self.inputContainerView
                         attribute:NSLayoutAttributeTrailing
                       multiplier:1.0
                         constant:-LAST_MSG_BUTTON_OFFSET
],
```

All colons are in one column (assuming a fixed width font which github apparently does not even use for code blocks).

**Comment by tmolitor-stud-tu** [comment]
Hmmm, your screen recording somehow looked like the position had changed between develop and your PR.

I'd go with your current PR then :)

**Pull request merged**